### PR TITLE
[SMTChecker] Output contract's name in counterexample

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: test/libsolidity/smtCheckerTests/functions/super_function_assert.sol [hunk 1]
```diff
@@ -28,5 +28,5 @@ contract D is C {
 	}
 }
 // ----
-// Warning 6328: (237-251): CHC: Assertion violation happens here.\nCounterexample:\nx = 2\n\n\n\nTransaction trace:\nconstructor()\nState: x = 0\nproxy()
-// Warning 6328: (360-374): CHC: Assertion violation happens here.\nCounterexample:\nx = 2\n\n\n\nTransaction trace:\nconstructor()\nState: x = 0\nproxy()
+// Warning 6328: (237-251): CHC: Assertion violation happens here.\nCounterexample:\nx = 2\n\n\n\nTransaction trace:\nC.constructor()\nState: x = 0\nA.proxy()
+// Warning 6328: (360-374): CHC: Assertion violation happens here.\nCounterexample:\nx = 2\n\n\n\nTransaction trace:\nD.constructor()\nState: x = 0\nA.proxy()
```
Comment (leonardoalt) [inline test/libsolidity/smtCheckerTests/functions/super_function_assert.sol:32]: These are nice

File: libsolidity/formal/Predicate.cpp [hunk 2]
```diff
@@ -183,7 +183,7 @@ string Predicate::formatSummaryCall(vector<smtutil::Expression> const& _args) co
 		fun->isFallback() ? "fallback" :
 		fun->isReceive() ? "receive" :
 		fun->name();
-	return fName + "(" + boost::algorithm::join(functionArgs, ", ") + ")";
+	return fun->annotation().contract->name() + "." + fName + "(" + boost::algorithm::join(functionArgs, ", ") + ")";
```
Comment (axic) [inline libsolidity/formal/Predicate.cpp:187]: Add assert for `fun->annotation().contract`?

**Comment by axic** [comment]
Perhaps worthy a changelog entry?

**Comment by leonardoalt** [comment]
Updated

File: libsolidity/formal/Predicate.cpp [hunk 3]
```diff
@@ -183,7 +183,8 @@ string Predicate::formatSummaryCall(vector<smtutil::Expression> const& _args) co
 		fun->isFallback() ? "fallback" :
 		fun->isReceive() ? "receive" :
 		fun->name();
-	return fName + "(" + boost::algorithm::join(functionArgs, ", ") + ")";
+	solAssert(fun->annotation().contract, "");
```
Comment (ekpyron) [inline libsolidity/formal/Predicate.cpp:186]: I guess, no free function can end up here, since it's always external calls?

**Review by ekpyron — APPROVED** [review]

File: libsolidity/formal/Predicate.cpp (hunk 3 above)
Comment (leonardoalt) [reply libsolidity/formal/Predicate.cpp:186]: So far, yes. That will change soon, but free functions will need some refactoring anyway, would do that then

**Pull request merged**